### ISSUE TITLE
type package depends on types of a package which is not in DefinitelyType.

### ISSUE THREAD

**Issue by tqma113** [DefinitelyTyped/DefinitelyTyped#50870]
When I maintain a ts package which is in `DefinitelyType` and depends on types of a package which is not in `DefinitelyType`. How can I set the dependency? 

Like `webpack@5` and `@types/webpack-manifest-plugin`. I wish `@types/webpack-manifest-plugin` depends on `webpack@5` but it depends on `@types/webpack` for now. If you install `@types/webpack-manifest-plugin` it will have `"@types/webpack": "*"`dependence, but I wish it has `"webepack": "^5.0.0"` dependence(maybe peerDependencee is better). How should I do?

I find all webpack plugin package has this issue but I find no solution.

**Comment by tqma113** [comment]
#50802

**Comment by tqma113** [comment]
#50773

**Comment by peterblazejewicz** [comment]
no clean or win-win solution IMO. 

> When I maintain a ts package which is in DefinitelyType and depends on types of a package which is not in DefinitelyType. How can I set the dependency

`package.json` and make this external package allowed on DT (refer to README)

The only other ways, `paths`. Can you make a gist that shows the problem (conflicting packages setup), thx!

**Comment by tqma113** [comment]
@peterblazejewicz thank you for your reply, let me have a try.

**Comment by peterblazejewicz** [comment]
another way:
I had an issue with an app, but only within IDE (VSCode), not wiht compiler.
To resolve 'which webpack' problem, I've restorted to this:
```ts
  "typeAcquisition": {
    "enable": true,
    "exclude": ["webpack"]
  },
  ```
 this removes resolution to `@types/webpack`, and explicitely always takes `node_modules/webpack` types.

**Comment by tqma113** [comment]
@peterblazejewicz 

> no clean or win-win solution IMO.
> 
> > When I maintain a ts package which is in DefinitelyType and depends on types of a package which is not in DefinitelyType. How can I set the dependency
> 
> `package.json` and make this external package allowed on DT (refer to README)
> 
> The only other ways, `paths`. Can you make a gist that shows the problem (conflicting packages setup), thx!
> 
> ```json
> "paths": {
>             "webpack": [
>                 "./node_modules/webpack"
>             ]
>         }
> ```
> 
> so to use `paths` as explicit fallback location (`do not look into 'types', look into ...`):
> https://www.typescriptlang.org/docs/handbook/module-resolution.html#path-mapping (section about fallbacks)

This solution will cause some error.
<details>
<summary>
Error log
</summary>
<pre>
<code>
Error: Errors in typescript@4.3 for external dependencies:
node_modules/webpack/types.d.ts(87,2): error TS2305: Module '"tapable"' has no exported member 'AsArray'.
node_modules/webpack/types.d.ts(1269,5): error TS2344: Type '"tap" | "tapAsync" | "tapPromise" | "name"' does not satisfy the constraint 'keyof AsyncSeriesHook<[Set<Chunk>], any, any>'.
  Type '"name"' is not assignable to type 'keyof AsyncSeriesHook<[Set<Chunk>], any, any>'.
node_modules/webpack/types.d.ts(1273,30): error TS2344: Type '"tap" | "tapAsync" | "tapPromise" | "name"' does not satisfy the constraint 'keyof AsyncSeriesHook<[], any, any>'.
  Type '"name"' is not assignable to type 'keyof AsyncSeriesHook<[], any, any>'.
node_modules/webpack/types.d.ts(1278,5): error TS2344: Type '"tap" | "tapAsync" | "tapPromise" | "name"' does not satisfy the constraint 'keyof AsyncSeriesHook<[Set<Chunk>], any, any>'.
node_modules/webpack/types.d.ts(1284,5): error TS2344: Type '"tap" | "tapAsync" | "tapPromise" | "name"' does not satisfy the constraint 'keyof AsyncSeriesHook<[Set<Chunk>], any, any>'.
node_modules/webpack/types.d.ts(2238,5): error TS2344: Type '"tap" | "tapAsync" | "tapPromise" | "name"' does not satisfy the constraint 'keyof AsyncSeriesWaterfallHook<[any[]], any, any>'.
  Type '"name"' is not assignable to type 'keyof AsyncSeriesWaterfallHook<[any[]], any, any>'.
node_modules/webpack/types.d.ts(6295,12): error TS2315: Type 'MultiHook' is not generic.
node_modules/webpack/types.d.ts(6296,8): error TS2315: Type 'MultiHook' is not generic.
node_modules/webpack/types.d.ts(6298,13): error TS2315: Type 'MultiHook' is not generic.
node_modules/webpack/types.d.ts(6299,22): error TS2315: Type 'MultiHook' is not generic.

    at /Users/tqma/Projects/github/DefinitelyTyped/node_modules/dtslint/bin/index.js:206:19
    at Generator.next (<anonymous>)
    at fulfilled (/Users/tqma/Projects/github/DefinitelyTyped/node_modules/dtslint/bin/index.js:6:58)
</code>
</pre>
</details>

**Comment by peterblazejewicz** [comment]
Yes, `tapable` now ships its own types as well.
Also there is `resolution` key in `package.json`, which helps to resolve webpack problems:
```json
"resolutions": {
    "webpack": "{version}"
 }
 ```
 would that help in your use-cases?

**Comment by tqma113** [comment]
Unfortunately, it can't pass the validation.

<details>
<summary>
Error log
</summary>
<pre>
<code>
▶ npm run test webpack-dev-middleware

> definitely-typed@0.0.3 test /Users/tqma/Projects/github/DefinitelyTyped
> dtslint types "webpack-dev-middleware"

Error: /Users/tqma/Projects/github/DefinitelyTyped/types/webpack-dev-middleware/package.json should not include field resolutions
    at Object.<anonymous> (/Users/tqma/Projects/github/DefinitelyTyped/node_modules/dtslint/bin/checks.js:51:27)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/tqma/Projects/github/DefinitelyTyped/node_modules/dtslint/bin/checks.js:5:58)
</code>
</pre>
</details>

**Comment by peterblazejewicz** [comment]
I remember I've metntioned that problem for MS team on Discord, maybe it's time for another shout, just to exchange ideas. No promises on the outcomes, though.

**Comment by tqma113** [comment]
@peterblazejewicz Can the issue be resolved at #44317 or https://gitter.im/DefinitelyTyped/DefinitelyTyped?

**Comment by peterblazejewicz** [comment]
@tqma113 
there is some effort to fix the DT problems:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/51712
(this is for DT project itself, not for client resolving dependencies).
We have discussion on this already on Discord, not on Gitter. TS team hangs on on Discord community channel.
There is a 'definitely'; typed thread: https://discord.gg/dTg6twt2 (if expired, just go to #definitely-typed)

We pointed possible solution we would like to use or see being used (explicit resolution of dependencies, etc, using either package.json or tsconfig.json, etc).

Note there are two kind of problems surfaced by native webpack brekaing changes:
- DT (this project) problems
- client (in the wild) problems of the same type (resolution of 'webpack' and '@types/webpack' dependencies)
